### PR TITLE
chore: make Autocomplete `@noResultsText` required

### PR DESCRIPTION
**Pull request by clintcs**
## 🚀 Description

Makes `@noResultsText` required as a followup to [this discussion](https://github.com/CrowdStrike/ember-toucan-core/pull/238#discussion_r1276486527).

- Makes Autocomplete `@noResultsText` required
- Adds "Optional" descriptors to Autocomplete's arguments documentation
- Add "No results text" to Autocomplete Field documentation
- Moves Multiselect's `@noResultsText` documentation to the top because it's required

## 🔬 How to Test

We're good as long as the build is green.

### CODE REVIEW

**Comment by clintcs** [comment]
Pushed a second commit with the following improvements: 

- Added "Optional" descriptors to Autocomplete's arguments documentation
- Added "No results text" to Autocomplete Field's documentation
- Moved documentation for required arguments to the top of Autocomplete and Multiselect

File: docs/components/autocomplete/index.md [hunk 1]
```diff
@@ -148,19 +182,21 @@ export default class extends Component {
...
-  handleFilter(value) {
+  onFilter(value) {
     return this.options.filter((option) => option === value);
   }
 }
 ```
 
 ## Disabled State
 
+Optional.
```
Comment (clintcs) [inline docs/components/autocomplete/index.md:198]: Unlike the arguments documentation above, the documentation here is for states that reference arguments—so they're very slightly different. @ynotdraw: Should these states have an "Optional" descriptor to match the arguments documentation? Or do you think of them differently? If the former, we might make a note to add "Optional" to other components' state documentation.

File: docs/components/multiselect-field/index.md [hunk 2]
```diff
@@ -198,6 +150,54 @@ Required.
 </Form::Fields::Multiselect>
 ```
 
+## Hint
```
Comment (clintcs) [inline docs/components/multiselect-field/index.md:153]: Moved down so the documentation for "No results text" is next to the other required arguments. @ynotdraw: Required-argument documentation at the top across the board? If you like, I can do an audit and then come back around to tweak documentation for the other components.

File: docs/components/autocomplete/index.md (hunk 1 above)
Comment (ynotdraw) [reply docs/components/autocomplete/index.md:198]: I can dig having "Optional" for these like you have in this PR 🎉

File: docs/components/multiselect-field/index.md (hunk 2 above)
Comment (ynotdraw) [reply docs/components/multiselect-field/index.md:153]: Sounds great to me. Yeah I think maybe moving the required to the top is likely best as if folks don't provide those arguments they'll run into issues. I think when we initially started we may have documented by personal preference / personal importance, but I think standardizing on maybe required first, then optional makes sense. I appreciate the thinking here!

**Review by ynotdraw — APPROVED** [review]

File: docs/components/multiselect-field/index.md (hunk 2 above)
Comment (clintcs) [reply docs/components/multiselect-field/index.md:153]: Will do!

File: docs/components/autocomplete/index.md (hunk 1 above)
Comment (clintcs) [reply docs/components/autocomplete/index.md:198]: Great! I'll open a followup.

**Pull request merged**